### PR TITLE
Add option to ignore custom resource tags

### LINKED ISSUE #140: Ignoring Custom Tags on TF Resources
Hi team,

are custom tags currently ignored by the TF reconciliation loop? Specifically: if I add a tag to a `subNet` created by the AWS extension provider.

Regards,
Andreas

/kind question

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area usability
/kind enhancement
/priority normal
/platform aws

**What this PR does / why we need it**:

This PR adds the option to ignore custom resource tags during infrastructure reconciliation, that were added by the end user or some automation, as proposed in #140.

It is forbidden by validation to ignore reserved tag keys (`Name`) and reserved tag key prefixes (`kubernetes.io`, `gardener.cloud`).

It also adds documentation for it as well as checks in the infra integration tests.

**Which issue(s) this PR fixes**:
Fixes #140

**Special notes for your reviewer**:

/cc @afritzler 

**Release note**:

```feature user
It is possible now to specify custom resource tags that should be ignored during infrastructure reconciliation (i.e. not removed) in the AWS `infrastructureConfig`. See [the documentation](https://github.com/gardener/gardener-extension-provider-aws/blob/master/docs/usage-as-end-user.md#infrastructureconfig) for more details.
```

### CODE REVIEW

**Comment by afritzler** [comment]
Awesome! Thank you @timebertt !!!

**Review by timuthy — APPROVED** [review]
/lgtm

File: pkg/controller/infrastructure/actuator_reconcile.go [hunk 1]
```diff
@@ -159,6 +162,12 @@ func generateTerraformInfraConfig(ctx context.Context, infrastructure *extension
 		enableECRAccess = *v
 	}
 
+	if infrastructureConfig.IgnoreTags != nil {
+		tags := infrastructureConfig.IgnoreTags.DeepCopy()
```
Comment (rfranzke) [inline pkg/controller/infrastructure/actuator_reconcile.go:166]: Why is this needed?

File: docs/usage-as-end-user.md [hunk 2]
```diff
@@ -166,6 +172,17 @@ If you want to use multiple availability zones then add a second, third, ... ent
 
 Apart from the VPC and the subnets the AWS extension will also create DHCP options and an internet gateway (only if a new VPC is created), routing tables, security groups, elastic IPs, NAT gateways, EC2 key pairs, IAM roles, and IAM instance profiles.
 
+The `ignoreTags` section allows to configure which resource tags on resources managed by Gardener should be ignored during
+infrastructure reconciliation. By default, all tags that are added outside of Gardener's / terraform's
+reconciliation will be removed during the next reconciliation. This field allows users and automation to add
+custom tags on resources created and managed by Gardener without loosing them on the next reconciliation.
+See [this document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/resource-tagging#ignoring-changes-in-all-resources)
+for details of the underlying terraform implementation.
```
[the remark this answers is not in the document]
Comment (timebertt) [reply docs/usage-as-end-user.md:180]: Yeah, you're right. Removed all references to terraform from the doc as this indeed shouldn't be relevant for end users.

File: pkg/controller/infrastructure/actuator_reconcile.go (hunk 1 above)
Comment (timebertt) [reply pkg/controller/infrastructure/actuator_reconcile.go:166]: Ups, left over. Removed it.

**Comment by timebertt** [comment]
Thanks for your suggestions, @rfranzke.
I've addressed them, PTAL.

/merge squash
/test

**Review by rfranzke — APPROVED** [review]
